lib/experimental/Information/Avatars/utils.tsx

```diff
@@ -0,0 +1,53 @@
+import { sizes } from "@/ui/avatar"
+import { ComponentProps } from "react"
+import { CompanyAvatar } from "./CompanyAvatar"
+import { PersonAvatar } from "./PersonAvatar"
+import { TeamAvatar } from "./TeamAvatar"
+
+type PersonAvatarProps = ComponentProps<typeof PersonAvatar>
+type TeamAvatarProps = ComponentProps<typeof TeamAvatar>
+type CompanyAvatarProps = ComponentProps<typeof CompanyAvatar>
+
+export type AvatarVariant =
+  | ({ type: "person" } & PersonAvatarProps)
+  | ({ type: "team" } & TeamAvatarProps)
+  | ({ type: "company" } & CompanyAvatarProps)
+
+export function renderAvatar(
+  avatar: AvatarVariant,
+  size: (typeof sizes)[number] = "xsmall"
+) {
```

Comment:
There's the risk that we add a new type of avatar (`avatar.type`) and we forget to deal with it.

Here's a trick:

```suggestion
export function renderAvatar(
  avatar: AvatarVariant,
  size: (typeof sizes)[number] = "xsmall"
): ReactNode {
```

By doing this, you're telling the TypeScript compiler that this component needs to *always* return something. Go and try to remove `case "team"` for example: it will complain.

Otherwise, without manually specifying, the return type of the function will be `ReactNode | undefined` which is not what we want!